lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1245,6 +1266,45 @@ selectCoinsForQuit ctx (ApiT wid) = do
   where
     genChange = delegationAddress @n
 
+{-------------------------------------------------------------------------------
+                                     Assets
+-------------------------------------------------------------------------------}
+
+listAssets
+    :: forall ctx s k.
+        ( ctx ~ ApiLayer s k
+        )
+    => ctx
+    -> ApiT WalletId
+    -> Handler [ApiAsset]
+listAssets ctx (ApiT wid) =
+    Handler $ ExceptT $ withDatabase df wid $ \db -> runHandler $ do
+        let wrk = hoistResource db (MsgFromWorker wid) ctx
```

Comment:
Use `withWorkerCtx` instead:

```hs
withWorkerCtx @_ @s @k ctx wid liftE liftE $ \wrk -> liftHandler $
        W.readWallet @_ @s @k wrk wid 
```

Reply:
`withWorkerCtx` comes with some error handling that is common to all wallets accesses.

Reply:
Thanks, that's better.